relay/client/manager.go

```diff
@@ -86,29 +88,66 @@ func (m *Manager) Serve() error {
 	if m.relayClient != nil {
 		return fmt.Errorf("manager already serving")
 	}
+	log.Debugf("starting relay client manager with %v relay servers", m.serverURLs)
 
-	m.relayClient = NewClient(m.ctx, m.serverURL, m.tokenStore, m.peerID)
-	err := m.relayClient.Connect()
-	if err != nil {
-		log.Errorf("failed to connect to relay server: %s", err)
-		return err
+	successChan := make(chan *Client, 1)
+
+	ctx, cancel := context.WithTimeout(m.ctx, connectionTimeout)
+	defer cancel()
+
+	sem := make(chan struct{}, maxConcurrentServers)
+
+	for _, url := range m.serverURLs {
+		sem <- struct{}{}
+		go func(url string) {
+			defer func() { <-sem }()
+			m.connect(m.ctx, url, successChan)
+		}(url)
 	}
 
-	m.reconnectGuard = NewGuard(m.ctx, m.relayClient)
-	m.relayClient.SetOnDisconnectListener(func() {
-		m.onServerDisconnected(m.serverURL)
-	})
-	m.startCleanupLoop()
+	// Wait for the first successful connection or all attempts to fail
+	select {
```

Comment:
If we still want to add additional transport methods (WS, UDP, custom TCP based things) to support the wide range of firewalls with the fastest way of communications, then this parallel connection attempt could cause overload on the server load balancer. I.e. the MGM server distributes 2 server addresses and we will support WS, UDP, custom TCP connections then every peer will send 6 connection attempts.